### PR TITLE
Fixes #4093: Additional status choices for vms

### LINKED ISSUE #4093: Status Decommissioning missing for virtual machines
### Environment
* Python version:  3.6.9
* NetBox version:   2.7.4

### Proposed Functionality
Currently a virtual machine can only have as status:
* Active
* Offline
* Staged

I'm missing here at least Decommissioning as status.

### Use Case
I would distinguish between Offline and Decommissioning in this way:

**Offline** is a VM that is temporary powered off and will be later powered on again.
**Decommissioning** is a VM that is powered off and will be later deleted.

Our decommissioning process is actually to power off a VM and then wait a reasonable time if still someone came up who needs this VM.
Being able to set Decommissioning as status would help to track VM's  planned for decommissioning.

Maybe it also makes sense to use for virtual machines the same status as for devices.
If you think of devices as hardware servers, the same processes are often used for hardware servers and VM's. Having the same status choices for both of them would help here.

### Database Changes
None

### External Dependencies
None

### PR DESCRIPTION
**Pull request by kobayashi**
### Fixes: #4093

When the issue is accepted, this PR can be merged.

### CODE REVIEW

File: netbox/virtualization/choices.py [hunk 1]
```diff
@@ -11,17 +11,20 @@ class VirtualMachineStatusChoices(ChoiceSet):
...
         (STATUS_OFFLINE, 'Offline'),
         (STATUS_STAGED, 'Staged'),
+        (STATUS_DECOMMISSIONING, 'Decommissioning'),
     )
 
     LEGACY_MAP = {
         STATUS_OFFLINE: 0,
         STATUS_ACTIVE: 1,
         STATUS_STAGED: 3,
+        STATUS_DECOMMISSIONING: 4,
```
Comment (jeremystretch) [inline netbox/virtualization/choices.py]: This is not needed. `LEGACY_MAP` exists only for backward-compatibility with pre-2.7 values; there is no need to populate new values.

File: netbox/virtualization/models.py [hunk 2]
```diff
@@ -267,9 +267,10 @@ class VirtualMachine(ChangeLoggedModel, ConfigContextModel, CustomFieldModel):
     ]
 
     STATUS_CLASS_MAP = {
-        'active': 'success',
-        'offline': 'warning',
-        'staged': 'primary',
+        VirtualMachineStatusChoices.STATUS_ACTIVE: 'success',
+        VirtualMachineStatusChoices.STATUS_OFFLINE: 'warning',
+        VirtualMachineStatusChoices.STATUS_STAGED: 'primary',
+        VirtualMachineStatusChoices.STATUS_DECOMMISSIONING: 'warning',
```
Comment (jeremystretch) [inline netbox/virtualization/models.py]: We should use different classes for the offline and decommissioning statuses. Might want to change offline to `danger` (red).

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/virtualization/choices.py (hunk 1 above)
Comment (kobayashi) [reply netbox/virtualization/choices.py]: Thx, removed the line.

File: netbox/virtualization/models.py (hunk 2 above)
Comment (kobayashi) [reply netbox/virtualization/models.py]: That is better. Changed the color of "OFFLINE" to `danger`

**Comment by jeremystretch** [comment]
@kobayashi Can you take a look at the latest comments on #4093 and let us know what you think about adding the additional statuses?

**Comment by kobayashi** [comment]
Set additional choices for virtual machine as the issue discussion.

File: netbox/virtualization/models.py (hunk 2 above)
Comment (kobayashi) [reply netbox/virtualization/models.py]: color label of vm status is set the same color as that of device.

**Pull request merged**